lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4196,7 +4267,10 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
         adjointPullbackStructArguments[origBB] = lastArg;
         continue;
       }
-      
+      // Add a pullback struct argument.
+      auto *pbStructArg = adjointBB->createPhiArgument(
```

Comment:
Adding `switch_enum` support exposed a latent bug here. Pullback struct argument should be created even if there are no dominated BB active values.

Reply:
Oops, I reverted the pullback struct argument reordering in https://github.com/apple/swift/pull/25509/commits/7459d4ec1c0a2af3c03ca7cfac78c666e14adce9 and couldn't reproduce the crash.